packages/next/src/build/webpack/plugins/minify-webpack-plugin/src/index.ts

```diff
@@ -18,27 +21,44 @@ function buildError(error: any, file: string) {
   }
 
   if (error.stack) {
-    return new Error(`${file} from Minifier\n${error.message}\n${error.stack}`)
+    return new WebpackError(
+      `${file} from Minifier\n${error.message}\n${error.stack}`
+    )
   }
 
-  return new Error(`${file} from Minifier\n${error.message}`)
+  return new WebpackError(`${file} from Minifier\n${error.message}`)
 }
 
 const debugMinify = process.env.NEXT_DEBUG_MINIFY
 
 export class MinifyPlugin {
+  constructor(private options: { noMangling?: boolean }) {}
+
   async optimize(
     compiler: any,
-    compilation: any,
+    compilation: Compilation,
     assets: any,
-    cache: any,
-    { SourceMapSource, RawSource }: any
+    cache: CacheFacade,
+    {
+      SourceMapSource,
+      RawSource,
+    }: {
+      SourceMapSource: typeof sources.SourceMapSource
+      RawSource: typeof sources.RawSource
+    }
   ) {
+    const mangle = this.options.noMangling ? false : true
```

Comment:
Huh? I assume there's a better idiom for boolean coercion in the nextjs codebase?